### PR TITLE
feat(common): import common utility logic

### PR DESCRIPTION
**Pull request by demetriusfeijoo**
## What?

- Adds a method that helps the creation of webhooks on the Storyblok side.
- Adds the `region-helper` package to the `nuxt-starter` and `story-starter` so it don't fail when cloned.
See here: https://github.com/storyblok/slack-plugin/pull/16#discussion_r1591572310

## Why?

So we can reuse these logics in other projects.

### CODE REVIEW

File: space-plugins/nuxt-base/server/utils/result.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { safeParse } from 'valibot';
+import {
+	ResultSchema,
+	type FailureResult,
+	type Result,
+	type SuccessResult,
+} from '../../types';
```
Comment (demetriusfeijoo) [inline space-plugins/nuxt-base/server/utils/result.ts:7]: It didn't work using the alias approach `~/types` that is the reason I used relative path here.

image

https://nuxt.com/docs/guide/going-further/layers#relative-paths-and-aliases

**Comment by eunjae-lee** [comment]
Hey @demetriusfeijoo I think there was a misunderstanding. What I suggested from [this ticket](https://storyblok.atlassian.net/browse/SHAPE-2807?focusedCommentId=88366) was to add only webhook logic here, not the error handling part, because we would want a proper discussion before adding it to the base layer.

**Comment by demetriusfeijoo** [comment]
> Hey @demetriusfeijoo I think there was a misunderstanding. What I suggested from [this ticket](https://storyblok.atlassian.net/browse/SHAPE-2807?focusedCommentId=88366) was to add only webhook logic here, not the error handling part, because we would want a proper discussion before adding it to the base layer.

Ohh that's true, @eunjae-lee. 
I forgot about it. 💀 

Do you think still would be nice to use valibot in the [webhook creation method](https://github.com/storyblok/space-tool-plugins/pull/45/files#diff-bd091b06f617be5fe50f5507f4ff563554b58dc4af566f1170aadd73baae0d6dR18)?

Currently, the valibot is being used by both the error-handling logic and the webhook, but as soon as I remove the error-handling from here, I don't know if using it only for the webhook creation would justify its use. What do you think? 🤔

**Comment by eunjae-lee** [comment]
> > Hey @demetriusfeijoo I think there was a misunderstanding. What I suggested from [this ticket](https://storyblok.atlassian.net/browse/SHAPE-2807?focusedCommentId=88366) was to add only webhook logic here, not the error handling part, because we would want a proper discussion before adding it to the base layer.
> 
> Ohh that's true, @eunjae-lee. I forgot about it. 💀
> 
> Do you think still would be nice to use valibot in the [webhook creation method](https://github.com/storyblok/space-tool-plugins/pull/45/files#diff-bd091b06f617be5fe50f5507f4ff563554b58dc4af566f1170aadd73baae0d6dR18)?
> 
> Currently, the valibot is being used by both the error-handling logic and the webhook, but as soon as I remove the error-handling from here, I don't know if using it only for the webhook creation would justify its use. What do you think? 🤔

I agree. Maybe we could write this function without valibot (as pure vanilla), and we could wrap it with valibot if we want later in actual projects. what do you think?

**Comment by demetriusfeijoo** [comment]
> I agree. Maybe we could write this function without valibot (as pure vanilla), and we could wrap it with valibot if we want later in actual projects. what do you think?

Yeah, I agree. 🙌 

The implementation would be something as:

```js
export const createStoryblokWebhook: CreateStoryblokWebhook = async (
	params,
) => {
	const isValid = validateWebhookCreationParams(params);
	
	if (!isValid) {
	    return; //or false?
	}
```

```js
const validateWebhookCreationParams = () => {
    // validation logic in vanilla js
}
```

Makes sense?

**Comment by eunjae-lee** [comment]
> > I agree. Maybe we could write this function without valibot (as pure vanilla), and we could wrap it with valibot if we want later in actual projects. what do you think?
> 
> Yeah, I agree. 🙌
> 
> The implementation would be something as:
> 
> ```js
> export const createStoryblokWebhook: CreateStoryblokWebhook = async (
> 	params,
> ) => {
> 	const isValid = validateWebhookCreationParams(params);
> 	
> 	if (!isValid) {
> 	    return; //or false?
> 	}
> ```
> 
> ```js
> const validateWebhookCreationParams = () => {
>     // validation logic in vanilla js
> }
> ```
> 
> Makes sense?

make sense to me! 

```
export const createStoryblokWebhook: CreateStoryblokWebhook = async (
	params,
) => {
	const isValid = isValidWebhookCreationParams(params);
	
	if (!isValid) {
	    return false
	}
```

```
const isValidWebhookCreationParams = () => {
    // validation logic in vanilla js
}
```

what do you think? ( i made slight changes)

File: space-plugins/nuxt-base/server/utils/webhook.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		return failure({
+			type: 'missing-parameters',
+			details: parsed.issues.map((issue) => issue.message),
+		});
+	}
+
+	const apiHost =
+		getRegion(params.spaceId) === US_CODE
+			? 'https://api-us.storyblok.com'
+			: 'https://mapi.storyblok.com';
```
[the remark this answers is not in the document]
Comment (BibiSebi) [reply space-plugins/nuxt-base/server/utils/webhook.ts:29]: yeah, we can have a separate ticket to test using region helper here, but would not see it as a blocker for now

**Review by BibiSebi — APPROVED** [review]
Looking good @demetriusfeijoo. Let's go 🚀  Please feel free to create a ticket for using the region helper and testing it out :)

**Comment by BibiSebi** [comment]
@Dawntraoz @eunjae-lee I am merging, but feel free to leave comments in case you spot something that needs to be addressed :)

**Pull request merged**